### PR TITLE
Assets revision

### LINKED ISSUE #408: [assets] Generate hashed assets to ensure cache busting on cdns
When running `gulp build --type production` it should generate 

```
app-<hash>.js
app-<hash>.css
```
and the elixir process needs to pic up those names to insert them correctly into the templates

cc @josevalim

### PR DESCRIPTION
**Pull request by milmazz**
Closes #408 

/cc @Dignifiedquire @josevalim

### CODE REVIEW

**Comment by milmazz** [comment]
@Dignifiedquire @josevalim How we handle the case that after execute `gulp build --type=production` now we get new files for `app-<hash>.{css,js}` from the Git point of view? Just delete the old files and add the new ones? wdyt?

**Comment by Dignifiedquire** [comment]
@milmazz I guess that's the only solution at the moment that I know :/

File: lib/ex_doc/formatter/html/templates/footer_template.eex [hunk 1]
```diff
@@ -13,6 +13,6 @@
   </div>
 </section>
 </div>
-    <script src="dist/app.js"></script>
+  <script src="<%= asset_rev Path.expand(config.output), "dist/app-*.js" %>"></script>
```
Comment (josevalim) [inline lib/ex_doc/formatter/html/templates/footer_template.eex]: Maybe we should call `Path.expand` inside `asset_rev`? :)

**Comment by josevalim** [comment]
:shipit:

File: lib/ex_doc/formatter/html/templates/footer_template.eex (hunk 1 above)
Comment (milmazz) [inline lib/ex_doc/formatter/html/templates/footer_template.eex]: Thanks for the feedback, fixed on master 62857db

**Pull request merged**